test/test_files/tck/match/match3.test

```diff
@@ -0,0 +1,415 @@
+                  (e {name: 'e'}), (f {name: 'f'}), (g {name: 'g'}), (h {name: 'h'}), 
+                  (i {name: 'i'}), (j {name: 'j'}), (k {name: 'k'}) 
+           CREATE (a)-[:KNOWS]->(d), (a)-[:KNOWS]->(e), (a)-[:KNOWS]->(f), (a)-[:KNOWS]->(g), 
+                  (a)-[:KNOWS]->(i), (b)-[:KNOWS]->(d), (b)-[:KNOWS]->(e), (b)-[:KNOWS]->(f), 
+                  (b)-[:KNOWS]->(h), (b)-[:KNOWS]->(k), (c)-[:KNOWS]->(d), (c)-[:KNOWS]->(e), 
+                  (c)-[:KNOWS]->(h), (c)-[:KNOWS]->(g), (c)-[:KNOWS]->(j);
+---- ok
+-STATEMENT MATCH (a {name: 'a'}), (b {name: 'b'}), (c {name: 'c'}) 
+           MATCH (a)-->(x), (b)-->(x), (c)-->(x) 
+           RETURN x;
+---- 2
+{_ID: 0:3, _LABEL: A, ID: 3, name: d}
+{_ID: 0:4, _LABEL: A, ID: 4, name: e}
+
+# Returning bound nodes that are not part of the pattern
+-CASE Scenario22
+-STATEMENT CREATE NODE TABLE A(ID SERIAL, name STRING, PRIMARY KEY(ID));
+---- ok
+-STATEMENT CREATE REL TABLE KNOWS(FROM A TO A);
+---- ok
+-STATEMENT CREATE (a {name: 'A'}), (b {name: 'B'}),
+                  (c {name: 'C'})
+           CREATE (a)-[:KNOWS]->(b);
+---- ok
+-STATEMENT MATCH (a {name: 'A'}), (c {name: 'C'})
+           MATCH (a)-->(b)
+           RETURN a, b, c;
+---- 1
+{_ID: 0:0, _LABEL: A, ID: 0, name: A}|{_ID: 0:1, _LABEL: A, ID: 1, name: B}|{_ID: 0:2, _LABEL: A, ID: 2, name: C}
+
+# Matching disconnected patterns
+-CASE Scenario23
+-STATEMENT CREATE NODE TABLE A(ID SERIAL, PRIMARY KEY(ID));
+---- ok
+-STATEMENT CREATE NODE TABLE B(ID SERIAL, PRIMARY KEY(ID));
+---- ok
+-STATEMENT CREATE NODE TABLE C(ID SERIAL,PRIMARY KEY(ID));
+---- ok
+-STATEMENT CREATE REL TABLE T1(FROM A TO B);
+---- ok
+-STATEMENT CREATE REL TABLE T2(FROM A TO C);
+---- ok
+-STATEMENT CREATE (a:A), (b:B), (c:C)
+           CREATE (a)-[:T1]->(b),
+                  (a)-[:T2]->(c);
+---- ok
+-STATEMENT MATCH (a)-->(b)
+           MATCH (c)-->(d)
+           RETURN a, b, c, d;
+---- 4
+{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 1:0, _LABEL: B, ID: 0}|{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 1:0, _LABEL: B, ID: 0}
+{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 1:0, _LABEL: B, ID: 0}|{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 2:0, _LABEL: C, ID: 0}
+{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 2:0, _LABEL: C, ID: 0}|{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 1:0, _LABEL: B, ID: 0}
+{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 2:0, _LABEL: C, ID: 0}|{_ID: 0:0, _LABEL: A, ID: 0}|{_ID: 2:0, _LABEL: C, ID: 0}
+
+# Matching twice with duplicate relationship types on same relationship
+-CASE Scenario24
+-STATEMENT CREATE NODE TABLE A(ID SERIAL, PRIMARY KEY(ID));
+---- ok
+-STATEMENT CREATE NODE TABLE B(ID SERIAL, PRIMARY KEY(ID));
+---- ok
+-STATEMENT CREATE REL TABLE T(FROM A TO B);
+---- ok
+-STATEMENT CREATE (:A)-[:T]->(:B);
+---- ok
+-STATEMENT MATCH (a1)-[r:T]->() 
+           WITH r, a1 
+           MATCH (a1)-[r:T]->(b2) 
+           RETURN a1, r, b2;
+---- error
+Binder exception: Bind relationship r to relationship with same name is not supported.
```

Comment:
This is probably where we are different from openCypher standard. Can u additionally comment in the test file as well. Something like `# Different result` so that we can quickly locate what are the statements to be fixed. Apply this change to match3&4